rasa/core/agent.py

```diff
@@ -557,47 +431,42 @@ async def parse_message_using_nlu_interpreter(
                 }
 
         """
-
-        processor = self.create_processor()
+        if not self.is_ready():
```

Comment:
Does it make sense to make this a decorator? Somewhat duplicate with `ensure_loaded_agent` in `server` 🤔